src/pages/displayAllRepositories/index.tsx

```diff
@@ -0,0 +1,68 @@
+import './App.css';
+import { useContext, useState, useEffect } from 'react';
+import { getCommitsByRepository, getRepositoriesByUser } from '../../service';
+import { NavBar } from '../../components/navbar';
+import { UserProfileContext } from '../../context/user';
+import { Card } from '../../components/card';
+import { IRepositoryProps } from '../../types/repositories';
+
+
+export const DisplayAllRepositories = () => {
+
+  const [repositories, setRepositories] = useState<IRepositoryProps[]>([]);
+  const [nickname, setNickName] = useState('');
+  const { user, isStorage, getStorageProfileUser, getProfileUserStored } = useContext(UserProfileContext);
+
+  
+  useEffect(() => {
+
+    getStorageProfileUser()
+
+    if (isStorage) {
```

Comment:
If the method `getStorageProfileUser` changes the state `isStorage` this conditional will not work since a state change needs to occur in different scope. if you try to validate the value of `isStorage` at this moment, the variable will still have the old value. I also think since you are listening to the variable in the dependency array in line 26, if the `getStorageProfuleUser()` change the `isStorage` state, you'll enter in an infinite loop because you are calling the method in line 19 (into useEffect).